PLATFORM-4759| disable concurrent cron jobs for close wiki script

harnash opened the pull request:
Disallow running multiple close wiki scripts in parallel.

harnash committed:
PLATFORM-4759| disable concurrent cron jobs for close wiki script

harnash committed:
PLATFORM-4759| added option to specify offset for close wiki script

harnash committed:
PLATFORM-4759| some tasks require extra space for dumps - lets increase that

r0bj commented on docker/maintenance/cronjob-template.yaml:
```diff
@@ -34,9 +35,11 @@ spec:
               requests:
                 cpu: "2"
                 memory: 800Mi
+                ephemeral-storage: "100Gi"
```
Just FYI:
Your numbers are little bit too optimistic, how do you think we have hosts with spare 100-400GB free space? On average, hosts have ~300GB for _all_ workloads data, images, etc. So in this case you would use one node exclusively only for your workload?
Could you be a little more conservative in that matter?

harnash replied:
I could but the dumps are quite large. I'm working on splitting this into smaller chunks and I will tune those when I know how much I need. This is quite temporary and should be removed in few days.

harnash committed:
PLATFORM-4759| split close wiki image archive into smaller chunks.

It looks like we are running out of space on wokers/pods when
dumping images of larger wikis. This should save some space.

harnash committed:
PLATFORM-4759| do not abort wiki close when there is no images to dump

harnash committed:
PLATFORM-4759| tune ephemeral storage for maintenance cron jobs

themech commented on docker/maintenance/cronjob-template.yaml:
```diff
@@ -34,9 +35,11 @@ spec:
               requests:
                 cpu: "2"
                 memory: 800Mi
+                ephemeral-storage: "200Gi"
```
@r0bj is this size acceptable?

harnash replied:
It is a compromise. 80Gi limit for each part will give roughly 160Gi of storage needed + some spare in case of some weird content.

r0bj replied:
I was thinking more about something around 50GB :). In general, having such a big ephemeral-storage request for your workload may cause trouble in scheduling workload, so your pod (created by this cronjob) may stay in `creating` state forever because it cannot fit to any node.
@harnash you mention 80GB for each part, let me understand it better:
Can we make those parts smaller (eg. 50GB)?
Can we process each part in sequence: create part, process part, upload part, whatever, go to next part? In this case we could make parts even smaller. 

So @themech we can go with 200GB but there is certain risk of not scheduling your pod. The bigger storage request the bigger the risk.

harnash replied:
@r0bj Yeah. We can make them smaller.

themech commented on extensions/wikia/WikiFactory/Close/maintenance.php:
```diff
@@ -149,33 +158,37 @@ public function execute() {
+									unlink( $source );
+
+									// SUS-6077 | move to a next wiki instead of failing the entire process
+									continue;
```
after you added an extra loop - should this `continue` do to the next tarFile or to the next wiki?

harnash replied:
Right. Fixed in 2dccc7d1ba0

harnash replied:
Why PHP does not have labeled loop statements? 🤔

harnash committed:
PLATFORM-4759| fix continue statement to progress through outer loop

harnash committed:
Change size limit for image backup to 50GB

harnash committed:
Revert ephemeral-storage settings for maintenance scripts
